### PR TITLE
Add easy CMake FetchContent install to README

### PR DESCRIPTION
**Pull request by GavinRay97**
Instructions for how to install with FetchContent

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -506,6 +507,30 @@ compatible with `-pedantic`.
    - If static linking, additionally link with dbghelp (`-ldbghelp`) on windows or lib dl (`-ldl`)
      on linux.
 
+
+### With CMake FetchContent
```
Comment (jeremy-rifkin) [inline README.md:511]: Can we make it more clear that there are two ways to do things here: Either the easy cmake way or the manual way. I wouldn't want anyone to get confused and think the cmake setup is required then the make and library path stuff on top.

**Review by jeremy-rifkin — COMMENTED** [review]
Thanks so much @GavinRay97 for the contribution! I'm very greatful for all the cmake contributions and everyone's patience with me - I really have no idea what I'm doing when it comes to cmake.

The changes look good to me, I'm adding a couple quick comments

File: README.md [hunk 2]
```diff
@@ -506,6 +507,30 @@ compatible with `-pedantic`.
    - If static linking, additionally link with dbghelp (`-ldbghelp`) on windows or lib dl (`-ldl`)
```
Comment (jeremy-rifkin) [inline README.md:507]: The summary here can be moved down to the manual section too

File: README.md (hunk 1 above)
Comment (GavinRay97) [reply README.md:511]: Sounds good to me, I probably ought to clarify that you still need to copy over the header into your project too, so I'll add that there as well 👍

Comment (jeremy-rifkin) [reply README.md:511]: Thanks!

**Review by GavinRay97 — COMMENTED** [review]
Adopt changes from review

**Review by jeremy-rifkin — APPROVED** [review]

**Comment by jeremy-rifkin** [comment]
Thank you! Is copying the header file over to the project required?

**Comment by GavinRay97** [comment]
Ah, it is not -- my IDE had not caught up to the CMake changes so I wasn't seeing them.
If you want an easy reproduction, you should be able to test it with these two files:

```cmake
# ./CMakeLists.txt
project(my_project LANGUAGES CXX)

include(FetchContent)
FetchContent_Declare(libassert
        GIT_REPOSITORY "https://github.com/jeremy-rifkin/libassert"
        GIT_TAG "v1.1"
)
FetchContent_MakeAvailable(libassert)

add_executable(my_project src/main.cpp)
target_link_libraries(my_project PRIVATE assert dl)

# Copy "compile_commands.json" to the root of the project
# (Not necessary, but makes clangd-based tools work in editors like VS Code with LLVM clangd extension)
add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD
        COMMAND ${CMAKE_COMMAND} -E copy_if_different
        ${CMAKE_CURRENT_BINARY_DIR}/compile_commands.json
        ${CMAKE_CURRENT_SOURCE_DIR}/compile_commands.json
)
```

```cpp
// ./src/main.cpp
#include <assert.hpp>

int main()
{
    DEBUG_ASSERT(1 == 2);
}
```

And then a build + run with:
```sh
$ cmake --build ./build --config Debug --target all
.... <snipped>

$ ./build/my_project
Debug Assertion failed at /home/user/projects/libassert-demo/src/main.cpp:5: int main():
    DEBUG_ASSERT(1 == 2);

Stack trace:
# 1 libc.so.6    ? __libc_start_call_main
# 2              ? __libc_start_main_alias_2
# 3 my_project   ? _start

Aborted
```

**Comment by jeremy-rifkin** [comment]
Thank you for the clarification!

**Pull request merged**